browser/components/newtab/RemoteAboutNewTab.jsm

```diff
@@ -81,11 +81,19 @@ let RemoteAboutNewTab = {
   },
 
   getSuggestions: Task.async(function* (message) {
-    let suggestion = yield SearchProvider.getSuggestions(message.target.browser, message.data);
-    message.target.sendAsyncMessage("NewTab:ContentSearchService", {
-      suggestion,
-      name: "Suggestions",
-    });
+    try {
+      let suggestion = yield SearchProvider.getSuggestions(message.target.browser, message.data);
+
+      // In the case where there is no suggestion available, do not send a message.
+      if (suggestion !== undefined) {
+        message.target.sendAsyncMessage("NewTab:ContentSearchService", {
+          suggestion,
+          name: "Suggestions",
+        });
+      }
+    } catch(e) {
+      Cu.reportError(e);
```

Comment:
on the remote side we're expecting an object, but i thought it would be safer to avoid sending a message altogether if the object is undefined. when we remove message passing this will go